evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -1513,16 +1385,25 @@ def test_component_graph_sampler_y_passes(mock_estimator_fit):
     pytest.importorskip(
         "imblearn.over_sampling", reason="Cannot import imblearn, skipping tests"
     )
-    # makes sure the y value from oversampler gets passed to the estimator, even though StandardScaler has no y output
+    # makes sure the y value from oversampler gets passed to the estimator
     X = pd.DataFrame({"a": [i for i in range(100)], "b": [i % 3 for i in range(100)]})
     y = pd.Series([0] * 90 + [1] * 10)
-    component_list = [
-        "Imputer",
-        "SMOTE Oversampler",
-        "Standard Scaler",
-        "Logistic Regression Classifier",
-    ]
-    component_graph = ComponentGraph.from_list(component_list)
+    component_graph = {
```

Comment:
Again, I thought this could be valuable to keep around even though the original test (does y value get passed even though edge from standard scaler is unclear?) is different; still valuable to double check that we are indeed getting the oversampled y :)

Reply:
Just a tad confused by `does y value get passed even though edge from standard scaler is unclear?`, we're not expecting it to be passed right?

Reply:
Before this PR, we expect that the oversampler's y value gets passed to the estimator because standard scaler doesn't produce a y value so the "most recent y" value the estimator sees is still the oversampler's y, but this isn't really clear using the list API.

Now that we're specifying edges, it's more clear that the estimator gets the oversampler's y edge, which would have meant that this test isn't necessary to confirm that anymore, but I wanted to keep it around anyways in the case that the estimator does not get the oversampler's y :)